components/search_results/search_results.jsx

```diff
@@ -179,6 +180,13 @@ class SearchResults extends React.Component {
         const noResults = (!results || results.length === 0);
         const searchTerms = this.props.searchTerms;
 
+        // to avoid loading icon showing infinitely, if the first page
+        // has results but no scroll, trigger the second page search
+        // to mark search as ended
+        if (this.props.searchPage === 0 && !noResults) {
```

Comment:
What happens if there are 0 results in the page 0?

Reply:
In that case the `isEnd` is set to `true` directly and the loading icon doesn't show at all

Reply:
Ok, this is super hard to track, and is using very distant code to archive this. If we change that behavior in the mattermost-redux repo, expecting to have it solved in the origin, we can break it without noticing it.

My suggestion is to add the `!this.props.isSearchAtEnd` check here too.

Reply:
Added!

Reply:
Thanks! :)